framework/src/st_things/things_stack/src/common/framework/things_data_manager.c

```diff
@@ -0,0 +1,2345 @@
+static char g_certificate_file_path[MAX_FILE_PATH_LENGTH] = { 0 };
+static char g_private_key_file_path[MAX_FILE_PATH_LENGTH] = { 0 };
+
+static char g_cloud_address[MAX_CLOUD_ADDRESS] = { 0 };
+
+static char *g_manufacturer_name;
+static char *g_setup_id;
+static bool is_artik;
+
+static int g_wifi_interface;
+static wifi_freq_e g_wifi_freq;
+
+static int g_ownership_transfer_method = 0;
+
+static struct hashmap_s *g_resource_type_hmap = NULL;	// map for resource types
+static struct hashmap_s *g_device_hmap = NULL;
+
+static bool is_there_cloud_data = false;
+static const char *origin_cloud_json_str = "{\n\
+		\"cloud\":    {\n\
+			\"address\":    \"52.40.216.160:5683\"\n\
+		},\n\
+		\"wifi\": {\n\
+			\"ssid\": \"unknown\"\n\
+		}\n\
+}";
+
+static wifi_manager_softap_config_s ap_config;
+static easysetup_connectivity_type_e es_conn_type = es_conn_type_none;
+static wifi_manager_ap_config_s g_homeap_info;
+
+typedef int8_t INT8;
+
+#define STD_CI_CLOUD_ADDRESS "52.40.216.160:5683"
+#define MAX_CI_CLOUD_ADDRESS 256
+static es_cloud_signup_s *gpst_cloud_data = NULL;
+static char g_ci_cloud_address[MAX_CI_CLOUD_ADDRESS] = { 0 };
+
+struct st_attribute_s const gstAttr_x_com_samsung_provisioninginfo[] = {
+	{
+		.key = "x.com.samsung.provisioning.targets",
+		.type = 9,
+		.mandatory = false,
+		.rw = 1
+	},
+	{
+		.key = "x.com.samsung.provisioning.owned",
+		.type = 0,
+		.mandatory = false,
+		.rw = 1
+	},
+	{
+		.key = "x.com.samsung.provisioning.easysetupdi",
+		.type = 3,
+		.mandatory = false,
+		.rw = 1
+	},
+	{
+		.key = "x.com.samsung.provisioning.reset",
+		.type = 0,
+		.mandatory = false,
+		.rw = 3
+	},
+	{
+		.key = "x.com.samsung.provisioning.stopAP",
+		.type = 0,
+		.mandatory = false,
+		.rw = 3
+	}
+};
+
+const struct st_attribute_s const gstAttr_x_com_samsung_accesspointlist[] = {
+	{
+		.key = "x.com.samsung.accesspoint.items",
+		.type = 9,
+		.mandatory = true,
+		.rw = 1
+	}
+};
+
+static const struct st_resource_type_s const gst_resource_types[] = {
+	{
+		.rt = "x.com.samsung.provisioninginfo",
+		.prop_cnt = 5,
+		.prop[0] = &gstAttr_x_com_samsung_provisioninginfo[0],
+		.prop[1] = &gstAttr_x_com_samsung_provisioninginfo[1],
+		.prop[2] = &gstAttr_x_com_samsung_provisioninginfo[2],
+		.prop[3] = &gstAttr_x_com_samsung_provisioninginfo[3],
+		.prop[4] = &gstAttr_x_com_samsung_provisioninginfo[4]
+	},
+	{
+		.rt = "x.com.samsung.accesspointlist",
+		.prop_cnt = 1,
+		.prop[0] = &gstAttr_x_com_samsung_accesspointlist[0]
+	}
+};
+
+static const struct st_resource_s const gstResources[] = {
+	{
+		.uri = "/sec/provisioninginfo",
+		.interface_types = {"oic.if.a"},
+		.resource_types = {"x.com.samsung.provisioninginfo"},
+		.if_cnt = 1,
+		.rt_cnt = 1,
+		.policy = 3
+	},
+	{
+		.uri = "/sec/accesspointlist",
+		.interface_types = {"oic.if.s"},
+		.resource_types = {"x.com.samsung.accesspointlist"},
+		.if_cnt = 1,
+		.rt_cnt = 1,
+		.policy = 3
+	}
+};
+
+bool util_str_copy(char **dest, const char *src)
```

Comment:
This function is used nowhere.
Let's delete it.

Reply:
deleted the function util_str_copy()